Library/OcSmbiosLib/SmbiosPatch.c

```diff
@@ -105,6 +105,60 @@ SmbiosGetOriginalStructureCount (
   return SmbiosGetStructureCount (mOriginalTable, mOriginalTableSize, Type);
 }
 
+STATIC
+BOOLEAN
+SmbiosHasValidOemFormFactor (
+  IN  APPLE_SMBIOS_STRUCTURE_POINTER  Original
+  )
+{
+  return Original.Raw != NULL
+    && SMBIOS_ACCESSIBLE (Original, Standard.Type17->FormFactor)
+    && Original.Standard.Type17->FormFactor != 0;
+}
+
+STATIC
+UINT8
+SmbiosGetFormFactor (
+  IN  OC_SMBIOS_DATA                  *Data,
+  IN  APPLE_SMBIOS_STRUCTURE_POINTER  Original
+  )
+{
+  BOOLEAN  IsAutomatic;
+
+  IsAutomatic = !Data->ForceMemoryFormFactor;
+
+  if (IsAutomatic) {
+    if (SmbiosHasValidOemFormFactor (Original)) {
+      //
+      // Try to use the original value if valid first.
+      //
+      return Original.Standard.Type17->FormFactor;
+    } else if (Data->MemoryFormFactor != NULL) {
```

Comment:
Please do not use else if after return. Always use just if. Here and below. This improves code readability.